adaptivemd/analysis/pyemma/emma.py

```diff
@@ -172,10 +172,26 @@ def execute(
 
         ty = trajs[0].types[outtype]
 
+
+        engines = []
+        for traj in trajectories:
+            if traj.engine not in engines:
+                engines.append(traj.engine)
+        
+        if len(engines) > 1:
+            trajs = []
+            for traj in trajectories:
+                trajs.append(os.path.join(traj.location, traj.types[outtype].filename))
+            trajectory_file_name = ''
+        else:
+            trajs = list(trajectories)
+            trajectory_file_name = ty.filename
+
+
         t.call(
```

Comment:
hmm, I see what this is necessary for: You want to use loaded and generated trajectories (so multiple engines) but assume that the generated trajectories have different names. 

Your way is one way to do it and not change the `_remote.py`. I usually try to avoid hacks and use parameters in ways they are not supposed to (even if it works). That works now, but when someone changes the `_remote.py` and does not know your hack, it might fail. E.g. a (resonable) check that the outtype is not empty.

I am also not sure if we need to make more restrictions on the output types in a project: Like now you assume, that if I use `protein` in the analysis that the engine from all trajs have the same _idea_ of `protein`. I think that makes sense, but we could check, if `stride` is the same and `selection`, too. Filename can be different of course.

Btw. getting list of unique `Engine` objects is easiest to get using `set`

```py
engines = set(traj.engine for traj in trajectories)
``` 

What about just changing `_remote` to accept directly the full path, like in your multi engine approach, just not to pass the additional traj_name parameter. I think I did this exactly to not create a second list, but why not.